packages/api/src/middleware/tracking.ts

```diff
@@ -0,0 +1,50 @@
+import sql from "sql-template-strings";
+import { ApiToken } from "../schema/types";
+import { db } from "../store";
+import Table from "../store/table";
+
+const flushDelay = 60 * 1000; // 60s
+
+class Tracker {
+  updateLastSeenTasks: Map<string, number> = new Map();
```

Comment:
Might wanna use something like `death` to flush these when the process is interrupted (same for stream info)
https://stackoverflow.com/a/31526978